graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/spring/NonFederatedSchemaAutoConfiguration.kt

```diff
@@ -33,16 +33,19 @@ import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Import
 import java.util.Optional
 
 /**
- * SpringBoot autoconfiguration for generating GraphQL schema.
+ * SpringBoot autoconfiguration for generating a non-federated GraphQL schema object.
+ * This will override the beans in [FederatedSchemaAutoConfiguration] if federation is disabled.
  */
 @ConditionalOnProperty(value = ["graphql.federation.enabled"], havingValue = "false", matchIfMissing = true)
 @Configuration
-class SchemaAutoConfiguration {
+@Import(GraphQLExecutionConfiguration::class)
+class NonFederatedSchemaAutoConfiguration {
```

Comment:
it seems weird to call it with `NonFederated` prefix as that would imply `Federation` is the default

Reply:
Yea, I didn't know what to call it because we have this format.

What do I call the bean above this that imports both of these expecting one of the returns a `GrapQLSchema` bean and then creates a `GraphQL` bean? 

![Screen Shot 2020-08-31 at 11 43 18 AM](https://user-images.githubusercontent.com/2446877/91754954-2d345600-eb7f-11ea-83ac-c5126b845e46.png)

Reply:
We have the follow config classes and they all are very close in name

* `GraphQLAutoConfiguration`
* `GraphQLSchemaConfiguration`
* `SchemaAutoConfiguraiton`

Reply:
We only had `GraphQLAutoConfiguration` (root) and `SchemaAutoConfiguration` before ;). Technically new `GraphQLSchemaConfiguration` is actually only creating `GraphQL` bean and corresponding handler. Schemas are created in those other two configs that are imported.

Wondering whether we should be combining some of those together (as they only contain few beans) but I do like the separation by conditional properties better....

Reply:
We can always rename now that we have the `4.x.x` cut